### PR TITLE
feat(utils.py): bump tiktoken dependency to 0.7.0 (gpt-4o token counting support)

### PR DESCRIPTION
**Pull request by ghost**
## Title

Adds support for gpt-4o token counting

## Relevant issues

- Fixes issue where token_counter was using cl100k_base for token counting gpt-4o calls 
- Fixes issue where user could not set custom tiktoken cache dir 

## Type

🆕 New Feature
🐛 Bug Fix

## Changes

- bump tiktoken dep 
- check for `CUSTOM_TIKTOKEN_CACHE_DIR` in env before using local tiktoken copy 

cc: @jeromeroussin

### CODE REVIEW

File: litellm/utils.py [hunk 1]
```diff
@@ -4035,7 +4035,10 @@ def openai_token_counter(
     """
     print_verbose(f"LiteLLM: Utils - Counting tokens for OpenAI model={model}")
     try:
-        encoding = tiktoken.encoding_for_model(model)
+        if "gpt-4o" in model:
+            encoding = tiktoken.get_encoding("o200k_base")
```
Comment (jeromeroussin) [inline litellm/utils.py:4039]: This shouldnt be needed

Comment (ghost) [reply litellm/utils.py:4039]: how else would we calculate tokens for gpt-4o vs. gpt3.5 turbo

i'm thinking about azure deployments here @jeromeroussin

Comment (jeromeroussin) [reply litellm/utils.py:4039]: You're thinking of the case when the deployment name isn't "gpt-4o"? There is no guarantee it is going to contain "gpt-4o" either... I thought that is what base_model was for?

Comment (ghost) [reply litellm/utils.py:4039]: > There is no guarantee it is going to contain "gpt-4o" either

yea that's what the next line is for 

>  I thought that is what base_model was for

yup that does help. the aim of the if-check is to provide a good default sanity check

**Pull request merged**